sycl/doc/extensions/KernelProperties/KernelProperties.asciidoc

```diff
@@ -461,14 +474,14 @@ new properties, for example `device_has_all_of` and `device_has_any_of`:
 device_has_any_of<device_has<aspect::fp16, device_has<aspect::fp64>>`.
 --
 
-. How should the `property_list` member variable behave with inheritance?
+. How should an embedded property list behave with inheritance?
 +
 --
 *UNRESOLVED*: The specification currently allows for a class to inspect the
-`property_list` member variable from its base class(es) and construct a new
-`property_list` member variable that applies to all call operators. Associating
-different properties with different call operators via inheritance has the
-potential to be confusing and would increase implementation complexity.
+property list embedded into its base class(es) and construct a new property
+list that applies to all call operators. Associating different properties with
+different call operators via inheritance has the potential to be confusing and
+would increase implementation complexity.
```

Comment:
Is this issue really unresolved, or can it be deleted?  Is anyone proposing that we need a way to declare different properties for each call operator in the case when a single function object defines more than one call operator?

Reply:
I don't think we should delete it (since somebody might ask the same question) but I'm happy to mark it "RESOLVED".

Reply:
Marking is "RESOLVED" is OK with me.

Actually, I don't quite understand why this is related to inheritance.  It is possible to define a named function object with two overloads of  `operator()` even without inheritance.

Reply:
Fixed in 73594f1f4a8618a2bfff3f041f0aee2131f87c5e.

Reply:
> Actually, I don't quite understand why this is related to inheritance. It is possible to define a named function object with two overloads of operator() even without inheritance.

The concern as I understand it is code like the below:

```c++
struct A
{
  void operator()(sycl::item<>) { }
  auto get(properties_tag) {
   return properties{ sub_group_size<8> };
  } 
}

struct B : public A
{
  void operator()(sycl::nd_item<>) { }
  auto get(properties_tag) {
    return properties{ sub_group_size<16> };
  }
}
```

The current specification says that properties are associated with all call operators of the struct, so both the `sycl::nd_item` and `sycl::item` overloads of `B` would use a sub-group size of 16.  If a developer wants to add functionality in `B` and ensure that they don't break functionality of `A`, then they should check whether `A` declares any `properties` and include those in the `properties` of `B`.